src/devices/telldus.c

```diff
@@ -0,0 +1,195 @@
+    }
+    crc &= 0xff;
+    return crc;
+}
+
+
+static int telldus_callback(bitbuffer_t *bitbuffer) 
+{
+    char time_str[LOCAL_TIME_BUFLEN];
+    uint8_t *bb;
+    unsigned int i;
+	uint8_t sensor_id;
+    uint8_t r_crc, c_crc;
+    uint8_t channel;
+	uint8_t humidity;
+    int tmp;
+    float temperature;
+    data_t *data;
+
+    /* Get the time */
+    local_time_str(0, time_str);
+
+    /* Correct number of rows? */
+    if (bitbuffer->num_rows != 2) {
+        if (debug_output) {
+            fprintf(stderr, "%s %s: wrong number of rows (%d)\n", 
+                    time_str, __func__, bitbuffer->num_rows);
+        }
+        return 0;
+    }
+
+    /* Correct bit length? */
+    if (bitbuffer->bits_per_row[1] != BITLEN) {
+        if (debug_output) {
+            fprintf(stderr, "%s %s: wrong number of bits (%d)\n", 
+                    time_str, __func__, bitbuffer->bits_per_row[0]);
+        }
+        return 0;
+    }
+
+    bb = bitbuffer->bb[1];
+
+    /* Correct start sequence? */
+    if (bb[0] != STARTBYTE) {
+        if (debug_output) {
+            fprintf(stderr, "%s %s: wrong start byte\n", time_str, __func__);
+		}
+        return 0;
+    }
+
+    /* If debug enabled, print the raw information */
+    if (debug_output) {
+        fprintf(stderr, "%s %s: recieved data = ", time_str, __func__);
+        for (i = 0; i < PACKETLEN; i++) {
+            fprintf(stderr, "%02x ",bb[i]);
+        }
+        fprintf(stderr, "\n");
+    }
+
+    /* Correct CRC? */
+    r_crc = bb[PACKETLEN - 1]; /* Last byte in the data */
+    c_crc = telldus_crc8_bit(INIT_CRC, bb, PACKETLEN - 1); /* First five bytes */
+	
+    if (r_crc != c_crc) {
+        if (debug_output) {
+            fprintf(stderr, "%s %s: CRC failed, calculated %x, received %x\n",
+                    time_str, __func__, c_crc, r_crc);
+        }
+        return 0;
+    }
+
+    /* Message validated, now parse the data */
+
+	/* Sensor id */
+	sensor_id = (bb[2] >> 4) & 0x0f;
+	/* Let's rename sensor 1000 to 1, and 0111 to 2 */
+	if(sensor_id == 0x08)
+		sensor_id = 0x01;
+	else
+		sensor_id = 0x02;
+
+    /* Channel */
+    channel = bb[1] & 0x07;
+	channel = (~channel & 0x07) + 1;
+
+    /* Temperature */
+    tmp = bb[PACKETLEN-4] & 0x0f;
+    tmp <<= 8;
+    tmp |= bb[PACKETLEN-3];
+    tmp = ~tmp & 0xfff;
+	if(tmp & 0x800) { /* if sign bit is 1 */
+		tmp &= 0x7ff;
+		tmp = ~tmp + 1;
+	}
+    temperature = tmp / 10.0f;
+
+    /* Humidity */
+	humidity = bb[PACKETLEN-2];
+	if (humidity == 0x00) { /* if there is no humidity sensor */
+		data = data_make("time",			"",				DATA_STRING,	time_str,
+						 "model",			"",				DATA_STRING,	"Telldus/Proove thermometer",
+						 "channel",			"Channel",		DATA_INT,		channel,
+						 "sensor", 			"Sensor id",	DATA_INT,		sensor_id,
+						 "temperature_C",	"Temperature",	DATA_FORMAT,	"%.1f C", DATA_DOUBLE, temperature,
+						 "humidity",		"Humidity",		DATA_STRING,	"N/A",
+						 NULL);
+	} else {
+		humidity = ~humidity;
```

Comment:
looks like your whole buffer is inverted? Use bitbuffer_invert() instead. https://github.com/merbanan/rtl_433/blob/master/include/bitbuffer.h#L47